File: fetch-cpp/fetch.js
// JS version of fetch program

async function fetchAsyncAwait() {
  var response = await fetch('data/foo.json');
  var data = await response.json();
  console.log(data.someProp);
}

function fetchPromise() {
  fetch('data/foo.json')
    .then(function(response) {
      return response.json();
    })
    .then(function(data) {
      console.log(data.someProp);
    });
}

File: fetch-cpp/fetch.cpp
// C++ version of fetch program

#include "fetch.hpp"

void fetchAsyncAwait() {
  Response response = await(fetch("data/foo.json"));
  JSON data = await(response.json());
  printf("%s\n", data["someProp"].str().c_str());
}

void fetchPromise() {
  fetch("data/foo.json")
    .then([](Response response) {
      return response.json();
    })
    .then([](JSON data) {
      printf("%s\n", data["someProp"].str().c_str());
    });
}
